src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -110,9 +110,12 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 	readonly onCommandInvalidated = this._onCommandInvalidated.event;
 	private readonly _onCurrentCommandInvalidated = this._register(new Emitter<ICommandInvalidationRequest>());
 	readonly onCurrentCommandInvalidated = this._onCurrentCommandInvalidated.event;
+	private readonly _onRequestWriteToTextArea = this._register(new Emitter<string>());
+	readonly onRequestWriteToTextArea = this._onRequestWriteToTextArea.event;
 
 	constructor(
 		private readonly _terminal: Terminal,
+		private readonly _isScreenReaderOptimized: boolean,
```

Comment:
Since we only want this to run on the renderer side, not the pty host, I think we should have a component that builds on `ICommandDetectionCapability` similar to `MarkNavigationAddon`